xbmc/addons/addoninfo/AddonInfoBuilder.cpp

```diff
@@ -483,62 +471,59 @@ bool CAddonInfoBuilder::ParseXMLExtension(CAddonExtensions& addonExt, const TiXm
     std::string name = attribute->Name();
     if (name != "point")
     {
-      cstring = attribute->Value();
-      if (cstring)
+      const std::string& value = StringUtils::StringOrEmpty(attribute->Value());
+      if (!value.empty())
       {
-        std::string value = cstring;
         name = "@" + name;
-        extension.push_back(std::make_pair(name, SExtValue(value)));
+        extension.emplace_back(std::make_pair(name, SExtValue(value)));
```

Comment:
Those push_back to emplace_back changes are ofc okay, but either this change should be mentioned in the commit message or (better) put into a separate commit.